openfl/utilities/optimizers/torch/fedprox.py

```diff
@@ -115,34 +172,53 @@ def step(self, closure=None):
             nesterov = group["nesterov"]
             mu = group["mu"]
             w_old = group["w_old"]
-            for p, w_old_p in zip(group["params"], w_old):
+
+            # Validate old weights for FedProx
+            self._validate_old_weights(mu, w_old)
+
+            # Apply proximal term when mu != 0
+            apply_proximal = w_old is not None
+
+            for i, p in enumerate(group["params"]):
                 if p.grad is None:
                     continue
+
                 d_p = p.grad
+
+                # Apply weight decay
                 if weight_decay != 0:
                     d_p = d_p.add(p, alpha=weight_decay)
```

Comment:
Similar case: `weight_decay` can be guaranteed to be `>=0.0` during initialization and this check can be avoided